github-runners/terraform/base/main.tf

```diff
@@ -0,0 +1,380 @@
+    source      = "./scripts/cleanup.sh"
+    destination = "/home/ubuntu/cleanup.sh"
+  }
+
+  # # Github runner installation
+  provisioner "remote-exec" {
+    inline = [
+      "sleep 60",
+      "export DEBIAN_FRONTEND=noninteractive",
+      "sudo apt update -y",
+      "sudo apt upgrade -y",
+      "sudo apt install git curl wget gnupg openssl net-tools -y",
+      # Download runner image
+      "mkdir actions-runner && cd actions-runner",
+      "curl -o actions-runner-linux-x64-${var.gh_runner_version}.tar.gz -L https://github.com/actions/runner/releases/download/v${var.gh_runner_version}/actions-runner-linux-x64-${var.gh_runner_version}.tar.gz",
+      "echo '${lookup(var.gh_runner_checksums, "linux_x86_64", "")} actions-runner-linux-x64-${var.gh_runner_version}.tar.gz' | shasum -a 256 -c",
+      "tar xzf ./actions-runner-linux-x64-${var.gh_runner_version}.tar.gz",
+      # configure runner
+      "./config.sh --url https://github.com/subspace --token ${var.gh_token} --unattended --name linux_x86_64 --labels 'self-hosted,Linux,x86_64' --work _work --runasservice",
+      "sudo ./svc.sh install",
+      "sudo ./svc.sh start",
+      "sudo ./svc.sh status",
+      # install monitoring
+      "sudo wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --non-interactive --nightly-channel --claim-rooms ${var.netdata_room} --claim-token ${var.netdata_token} --claim-url https://app.netdata.cloud",
+
+    ]
+
+    on_failure = continue
+
+  }
+
+  # Setting up the ssh connection
+  connection {
+    type        = "ssh"
+    host        = element(self.*.public_ip, count.index)
+    user        = "ubuntu"
+    private_key = file("${var.public_key_path}")
+    timeout     = "90s"
+  }
+
+}
+
+resource "aws_instance" "linux_arm64_runner" {
+  count             = length(var.public_subnet_cidrs)
+  ami               = data.aws_ami.ubuntu_arm64.image_id
+  instance_type     = element(var.instance_type_arm, 0)
+  subnet_id         = element(aws_subnet.public_subnets.*.id, count.index)
+  availability_zone = element(var.azs, count.index)
+  # Security Group
+  vpc_security_group_ids = ["${aws_security_group.allow_runner.id}"]
+  # the Public SSH key
+  key_name                    = var.aws_key_name
+  associate_public_ip_address = true
+
+  tags = {
+    name       = "gh-linux-arm64-runner"
+    role       = "runner"
+    os_name    = "ubuntu"
+    os_version = "22.04"
+    arch       = "arm64"
+  }
+
+  depends_on = [
+    aws_subnet.public_subnets,
+    aws_nat_gateway.nat_gateway,
+    aws_internet_gateway.gw
+  ]
+
+  lifecycle {
+
+    create_before_destroy = true
+
+  }
+
+  provisioner "file" {
+    source      = "./scripts/secure.sh"
+    destination = "/home/ubuntu/secure.sh"
+  }
+
+  provisioner "file" {
+    source      = "./scripts/cleanup.sh"
+    destination = "/home/ubuntu/cleanup.sh"
+  }
+
+  # Github runner installation
+  provisioner "remote-exec" {
+    inline = [
+      "sleep 120",
```

Comment:
Can you add some explanation on why we need to sleep here ?

Reply:
It gives time for cloud-init to complete before running remote provisioner